recipe/meta.yaml

```diff
@@ -11,21 +11,20 @@ source:
 
 build:
     number: 0
-    features:
-        - vc9  # [win and py27]
-        - vc10  # [win and py34]
-        - vc14  # [win and py>=35]
-    skip: true  # [win and py>35]
+    run_exports:
+        - {{ pin_subpackage('zlib', max_pin='x.x.x') }}
```

Comment:
You might want to point to https://abi-laboratory.pro/tracker/timeline/zlib/ to justify why it's pinned like this.

The biggest problem I have found with conda build 3 is that there is no easy way of knowing which packages (in your dependencies) have `run_exports` and which ones don't. Most C/C++ packages have them but there are exceptions. For example `boost` from the defaults channel does not have one. You could imagine people who want to create a recipe would be really confused about which of their dependencies have `run_exports`.

Reply:
Would some kind of debugging output from conda-build help?

Our recipes in defaults do always have comments like this explaining the pin value: https://github.com/AnacondaRecipes/zlib-feedstock/blob/master/recipe/meta.yaml#L17

Reply:
I think the best solution would be to have this information in the channel index so that conda search can be used. This will allow us to see for example which versions of a package have run exports or to see which of the boost packages (cpp-boost, py-boost) have run exports.

Reply:
I don't think it's wise to have it in the index itself.  We want to be extremely careful about adding stuff to the index.  For maximum efficiency, it needs to be only what is necessary at install time.  This is not to say that there can't be a supplemental index-like aggregation of run_index values to some central file - just that it should not be in repodata.json.

Reply:
Yes that makes sense. As long as there is some search-able index per-channel to find these information, it should be fine. Maybe conda channels can have an additional data file which would contain these kind of extra information.